tools/hubl/internal/registry.go

```diff
@@ -0,0 +1,104 @@
+package internal
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/manifoldco/promptui"
+)
+
+type ChainRegistryEntry struct {
+	APIs struct {
+		GRPC []*APIEntry `json:"grpc"`
+	} `json:"apis"`
+}
+
+type APIEntry struct {
+	Address  string
+	Provider string
+}
+
+func GetChainRegistryEntry(chain string) (*ChainRegistryEntry, error) {
+	res, err := http.Get(fmt.Sprintf("https://raw.githubusercontent.com/cosmos/chain-registry/master/%v/chain.json", chain))
+	if err != nil {
+		return nil, err
+	}
+	bz, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = res.Body.Close(); err != nil {
+		return nil, err
+	}
+
+	data := &ChainRegistryEntry{}
+	if err = json.Unmarshal(bz, data); err != nil {
+		return nil, err
+	}
+
+	// clean-up the URL
+	cleanEntries := make([]*APIEntry, 0)
+	for i, apiEntry := range data.APIs.GRPC {
+		// clean-up the http(s):// prefix
+		if idx := strings.Index(apiEntry.Address, "://"); idx != -1 {
+			data.APIs.GRPC[i].Address = apiEntry.Address[idx+3:]
+		}
+
+		// remove trailing slashes
+		data.APIs.GRPC[i].Address = strings.TrimSuffix(data.APIs.GRPC[i].Address, "/")
+
+		// remove addresses without a port
+		if !strings.Contains(data.APIs.GRPC[i].Address, ":") {
+			continue
+		}
+
```

Comment:
I believe using the `net/url` would be safer here. Also please add unit tests.

I think that a string such as `://://somerubbish:abcd//123//` will be successfully stored as `://somerubbish:abcd//123`. With the url package it would fail. In the end I think you would probably only want to keep the host from the address (playground: https://go.dev/play/p/X74iDrAnTTp) + perhaps `path+rawQuery+fragment` depending on the intent